pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -88,3 +92,133 @@ type ServiceExportList struct {
 	// +listType=set
 	Items []ServiceExport `json:"items"`
 }
+
+// ServiceExportConditionType is a type of condition associated with ServiceExport.
+// This type should be used with the ServiceExportStatus.Conditions
+// field.
+type ServiceExportConditionType string
+
+// ServiceExportConditionReason defines the set of reasons that explain
+// why a particular conflict condition type has been raised.
+type ServiceExportConditionReason string
```

Comment:
The types don't really add any value b/c the Condition fields are still string and it forces a type conversion to string. In fact, we used to have `ServiceExportConditionType` when we had `ServiceExportCondition` but it was [removed](https://github.com/kubernetes-sigs/mcs-api/pull/33) when we switched to `metav1.Condition`.

Reply:
Well Gateway-API does that and indeed implementations need a cast to string everywhere, I don't know if we want to actually do that or not

Reply:
I think that would probably makes sense in a context where you have some internal processing in the implementations and want to type your function rather than returning a plain string.

Although we don't do that in Cilium for instance so for us too the value of keeping that would be pretty much none (while requiring to cast in many places). I am guessing it would be pretty much the same for you @tpantelis?

Reply:
Yes - the type casting became too much. Now we could provide a helper function like NewServiceExportValidCondition(reason ServiceExportConditionReason …) to hide the type casts.  But, in the end, any custom string not defined here is still valid so the type doesn’t really add value.

Reply:
Ok then I removed those, we could still reconsider this if we find some other benefits

Reply:
> I think that would probably makes sense in a context where you have some internal processing in the implementations and want to type your function rather than returning a plain string.

Yea, precisely this and avoiding accidentally returning a condition of the wrong type (e.g. a ServiceImport reason for a ServiceExport type). In practice I think the typecasts aren't particularly annoying (and I believe are elided in compilation), here's an [example](https://github.com/istio/istio/blob/34895c89f0e44758da6f6d22d8b8f56754fd4f31/pilot/pkg/config/kube/gateway/conditions.go#L147-L156) from Istio's Gateway API implementation.

I don't think they're essential though, just a minor detail to guide implementers of the spec.

Reply:
Yeah the types can make sense when used in functions. Looking at the Submariner code, we do actually have a function `newServiceExportCondition(condType string, status metav1.ConditionStatus, reason, msg string) metav1.Condition` so we could simply change the signature. It's the unit tests which have more type casting.

If we want to define types then I think it would be helpful to provide a `NewServiceExportCondition(condType ServiceExportConditionType, status metav1.ConditionStatus, reason ServiceExportConditionReason, msg string) metav1.Condition` function that uses the types to further guide implementors.

Reply:
Added as suggested!